File: src/WaylandClient_1_connect.pas
// By Fred vS 2023.

program WaylandClient_1_connect;

{$mode objfpc}{$H+}
{$packrecords c}
{$interfaces corba}

uses
  ctypes,
  wayland_client_core,
  SysUtils;

var
  display: pwl_display;

begin
  display := wl_display_connect(nil);
  if display <> nil then
  begin
    writeln('Connected!');
  end
  else
  begin
    writeln('Error connecting ;(');
    ExitCode := 1;
    Exit;
  end;

  wl_display_disconnect(display);
  ExitCode := 0;
end.

File: src/c/WaylandClient_1_connect.c
/*
   From https://wayland-book.com/xdg-shell-basics/example-code.html
   Compile with: 
   $ gcc -o connect WaylandClient_1_connect.c -lwayland-client
*/
#include <stdio.h>
#include <wayland-client.h>

int main(void)
{
    struct wl_display *display = wl_display_connect(NULL);
    if (display) {
        printf("Connected!\n");
    } else {
        printf("Error connecting ;(\n");
        return 1;
    }

    wl_display_disconnect(display);
    return 0;
}


